packages/website/src/features/Deploy/QueueFromGitOpsPage.tsx

```diff
@@ -474,113 +516,247 @@ export default function QueueFromGitOps() {
       }
     }
 
-    doBuild(cannonDefInfo?.def, prevCannonDeployInfo.pkg);
-  };
+    doBuild(cannonDefInfo?.def, partialDeployInfo?.ipfsQuery.data?.deployInfo);
+  }, [
+    isConnected,
+    openConnectModal,
+    chainId,
+    currentSafe?.chainId,
+    switchChainAsync,
+    doBuild,
+    cannonDefInfo?.def,
+    partialDeployInfo?.ipfsQuery.data?.deployInfo,
+    toast,
```

Comment:
Regardless the warning from the linter, which is just a reminder, everything should not be put on the hook state change array, e.g `openConnectModal` is not necessary and maybe triggers unnecessary re-renders.